mb_modelbase/server/modelbase.py

```diff
@@ -290,15 +305,35 @@ def execute(self, query):
         # basic syntax and semantics checking of the given query is done in the _extract* methods
         if 'MODEL' in query:
             base = self._extractFrom(query)
-            # maybe copy
-            derived_model = base if base.name == query["AS"] else base.copy(query["AS"])
-            # derive submodel
-            derived_model.model(
+
+            key = model_key(
+                name=query["AS"],
                 model=self._extractModel(query),
-                where=self._extractWhere(query),
-                default_values=self._extractDefaultValue(query),
-                default_subsets=self._extractDefaultSubset(query),
-                hide=self._extractHide(query)),
+                where=self._extractWhere(query)
+            )
```

Comment:
Reading this makes we wonder if the computation of the key should be part of the cache submodule?

Reply:
Right, they shouldn't be part of the cache submodule. The key related functions have been moved to the ModelBase class, where all the _extract... functions are located.